packages/core/chrome/core-chrome-browser-internal/src/ui/header/header.tsx

```diff
@@ -168,6 +168,14 @@ export function Header({
           <EuiHeader position="fixed" className="header__secondBar">
             <EuiHeaderSection grow={false}>
               <EuiHeaderSectionItem border="right" className="header__toggleNavButtonSection">
+                <EuiScreenReaderLive focusRegionOnTextChange>
+                  {i18n.translate('core.ui.primaryNav.screenReaderAnnouncement', {
+                    defaultMessage: '{title}',
+                    values: {
+                      title: application.currentAppId$ as unknown as string,
```

Comment:
Unfortunately this isn't quite the right var to use it looks like - it's an object, not a string:

<img width="731" alt="" src="https://user-images.githubusercontent.com/549407/220430109-83d8785b-fac2-4762-8506-26fb15c20c20.png">

It also won't react to navigation _within_ applications, only navigation _between_ applications. Ideally we really want to announce every page/route change just like a non-SPA website would.

I'll dig into this to see what Kibana's APIs can do, and push up some code to this branch.

Reply:
In case it helps, I did a brief investigation to learn how to react to `document.title` changes and documented them here: https://github.com/elastic/kibana/issues/149249#issuecomment-1420662078

cc @rshen91

Reply:
Ah! @cee-chen I think the scope of this PR is just for between applications but a subsequent PR can address within applications https://github.com/elastic/kibana/issues/135760#issuecomment-1396940435 for reference

Reply:
That's super help context, thank you Alejandro and Rachel!

My 2c: I think observing the `title` node is the way to go (it mimics non-SPA web navigation) and is worth doing right now as opposed to later. It shouldn't take too long to do, and I'm happy to help with that work here in this PR.

Reply:
Oof - this ended up being fair more complex than I thought unfortunately, for several reasons:

1. Sadly, _many_ plugins/apps are not correctly using `chrome.docTitle.change` (e.g. Maps, Analytics, Sample Data, the list goes on and on...) - so watching the `title` node isn't as helpful as it should be. Almost all apps _are_ correctly setting breadcrumbs, however, so I went with an approach of using the `breadcrumbs$` observable and flipping + joining the breadcrumbs array to output a string based on the page breadcrumbs.

2. I also went back and forth on focus region a bit - between apps, it makes sense that focus should reset to right before the primary collapsible nav. Inside apps, it makes more sense for focus to reset to the `main` content and skip any secondary navs. I ended up splitting the difference and deferring back to a tried and true accessibility paradigm of having focus reset to the top of the document and adding a skip link button that jumps to the `main` content which is the first focus target of the page. I think this works quite well, but will need @1Copenut to test on various SRs to confirm.
Note on the above: several apps not using either Eui or KibanaPageTemplate will not have `main`s to skip to, and will need to either rewrite their layout slightly or Kibana will need to add some API (similar to `chrome.docTitle.change` - maybe `chrome.skipLinkId.set`?) that allows consumers to set a custom ID of the region that keyboard users can skip to.

Let me know if you have any q's! BTW, I def recommend testing on VO+Safari and not VO+Firefox - interestingly enough, VO+FF attempts to aggressively read out `document.title` changes when Safari does not, but Safari is the more "correct" (or popular) usage.

Reply:
Thanks for the tip about testing on safari vs firefox! I'll definitely do that moving forward 👍🏻

Reply:
I'll give this a pass first thing Thursday morning. I had to clear a few backlog items today so I can give this my full attention. Very excited about these updates and the improvement they're going to bring!